tools/update_lint_rules/lib/src/services/lint_rule_service.dart

```diff
@@ -133,6 +138,6 @@ class LintRuleService {
             RuleState.deprecated || RuleState.removed => false,
           },
         );
-    return rules;
+    return _cacheAllRules = rules;
   }
 }
```

Comment:
![imo-badge](https://img.shields.io/badge/review-imo-skyblue.svg) 
I think it's a matter of preference, but I personally recommend an implementation using `AsyncCache` as it eliminates the need to handle `null`. Whether to make this change is up to you. However, if you decide to go with this modification, please be aware that you will need to execute `dart pub add async` to introduce the package.

```dart
final _allRulesCache = AsyncCache<Iterable<Rule>>(const Duration(minutes: 5));

@visibleForTesting
Future<Iterable<Rule>> getRules() => _allRulesCache.fetch(
      () async {
        final url = Uri.https(
          'raw.githubusercontent.com',
          'dart-lang/sdk/main/pkg/linter/tool/machine/rules.json',
        );

        final responseBody = await _appClient.read(url);

        final json = jsonDecode(responseBody) as List<dynamic>;

        final rules = json.map((e) => Rule.fromJson(e)).where(
              (e) => switch (e.state) {
                RuleState.stable || RuleState.experimental => true,
                RuleState.deprecated || RuleState.removed => false,
              },
            );
        return rules;
      },
    );
```

FYI:
https://pub.dev/documentation/async/latest/async/AsyncCache-class.html

Reply:
Thank you for pointing that out!
Rather than using AsyncCache, which specifies a rough Duration, I decided to use AsyncMemoizer because I thought it would be simpler and easier to understand. I would like to know if there are any problems with this implementation.

7792cb81e0db9d4cd385a3eac87cf9fc517d9426
12fd61ccfad539f97fb737275293005c9856b2da

Reply:
I didn't know there was such a thing as `AsyncMemoizer`! Thank you for informing me!